src/local_scheduler/local_scheduler_algorithm.cc

```diff
@@ -927,6 +927,19 @@ void give_task_to_local_scheduler(LocalSchedulerState *state,
   task_table_add_task(state->db, task, &retryInfo, NULL, state);
 }
 
+void give_task_to_global_scheduler_retry(UniqueID id,
+                                         void *user_context,
+                                         void *user_data) {
+  LocalSchedulerState *state = (LocalSchedulerState *) user_context;
+  Task *task = (Task *) user_data;
+  CHECK(Task_state(task) == TASK_STATUS_WAITING);
+
+  TaskSpec *spec = Task_task_spec(task);
+  CHECK(ActorID_equal(TaskSpec_actor_id(spec), NIL_ACTOR_ID));
+  handle_task_submitted(state, state->algorithm_state, spec,
```

Comment:
interesting choice. So you want to recirculate the task through the local scheduler's new task handling mechanism, even though the decision has already been made to forward it to the global scheduler? So the assumption is that Redis is busy, so let's try to keep it local again.

Reply:
The assumption is not that Redis is busy, but rather that the global scheduler has not started up yet.

We could alternatively simple call `give_task_to_global_scheduler` again.